.github/workflows/ci-commit.yml

```diff
@@ -0,0 +1,345 @@
+name: ci
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: ${{ matrix.os }}-${{  matrix.arch }}-${{ matrix.build }}
+    runs-on: ${{ matrix.os }}-${{  matrix.arch }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: 
+          - ubuntu-22.04
+          - ubuntu-23.10
+          - debian-11
+          - debian-12
+          - alpine-3.19
+          - rhel-9
+          - amazonlinux-2023
```

Comment:
There are more package lists than what's listed here, is that intentional?

Reply:
Yes and no: there is `alpine-3.18` list which I failed to use because there is no alpine 3.18 image for self hosted runners. So this one can be removed.
At the same time, having general lists (`debian`, `alpine`, `ubuntu`, `rhel`) is intentional: one can install a package for all versions of an OS family.

Reply:
rhel-8 also seems to be in the lists but not the workflow file...